core/primitives/impl/scale_codec_impl.cpp

```diff
@@ -236,27 +266,19 @@ namespace kagome::primitives {
     OUTCOME_TRY(extrinsics_root, decoder.decode(stream));
     OUTCOME_TRY(digest, decodeCollection<uint8_t>(stream));
 
-    return BlockHeader(parent_hash, number, state_root,
-                       extrinsics_root, Buffer{digest});
+    return BlockHeader(parent_hash, number, state_root, extrinsics_root,
+                       Buffer{digest});
```

Comment:
I assume every argument can be moved into the ctor. Except for `number`, as for it there's no sense in moving

Reply:
```suggestion
                       Buffer(std::move(digest)));
```
I'm not sure, but it looks more logical.